seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/common/serialization/InternalRowConverter.java

```diff
@@ -164,11 +168,11 @@ private static Object reconvert(Object field, SeaTunnelDataType<?> dataType) {
             case ROW:
                 return reconvert((InternalRow) field, (SeaTunnelRowType) dataType);
             case DATE:
-                return ((Date) field).toLocalDate();
+                return LocalDate.ofEpochDay((int) field);
```

Comment:
The same as above.

Reply:
how about like this? `(int) ((Date.valueOf(String.valueOf(val))).toLocalDate().toEpochDay());`

Reply:
> (int) ((Date.valueOf(String.valueOf(val))).toLocalDate().toEpochDay());
 the value is int in sparkRow, and  transfrom it to LocalDate.

Reply:
> how about like this? `(int) ((Date.valueOf(String.valueOf(val))).toLocalDate().toEpochDay());`
the date value in seatunal is LocalDate,we don't need to transform  like `obj->string->date->localdate`